src/components/Tracks/Tracks.js

```diff
@@ -31,10 +33,9 @@ const Tracks = () => {
         {tracks?.map((track) => (
           <Track
             key={track.id}
-            name={track.name}
-            albumImage={track.albumImage}
-            duration={track.duration}
-          ></Track>
+            selected={currentTrack?.id === track.id ?? false}
+            track={track}
+          />
         ))}
```

Comment:
When a user clicks on a track, this track becomes the current one (when are gonna play this track via audio player). And we also want to make this track _selected_ in the list.
1. `<Track>` component is only responsible for dispatching the current track. A track becomes _current_ on user's click.
2. `<Tracks>` component gets notified when the current track is changed. This is only needed to specify which track is _selected_ in the list. 

So, do we reevaluate every single `<Track>`  when the currrent track is changed? In fact no, because we use React.memo for `<Track>` component.

Reply:
Too complicated rendering with lots of `ifelse`...

Reply:
Also, you can manage all Track events here. What is the point to split logic over 2 components?